alacritty_terminal/src/grid/mod.rs

```diff
@@ -508,11 +540,36 @@ impl<T: GridCell + Copy + Clone> Grid<T> {
         }
     }
 
+    pub fn clear_viewport(&mut self, template: &T) {
+        for i in 0..self.num_lines().0 {
+            self[i].occ = self.num_cols().0;
+        }
```

Comment:
I'm slightly concerned about this, since it will leave the `occ` at way too high values after each clear.

Reply:
Would it make sense to reset them back to 0 at the end of `clear_viewport`? As I'm understanding things right now, setting `occ = self.num_cols()` is telling the grid to consider each row as fully occupied for the purposes of clearing the viewport. And after a clear it should be completely empty.

I've tested this, and it seems to work.

Reply:
Actually I'm not sure if that will do anything. I have to investigate again what exactly is going on, but in theory if we set the occ to num_cols before scrolling, the stuff with incorrect occ should be what has been scrolled offscreen.

So I'm a bit confused about what exactly is happening. First thing we do is reset the occ, then we rotate, meaning that these lines are gone and then we reset the fixed lines (which I think should be what has been 'rotated in' from the bottom). So I don't quite get why the reftest failed exactly and I'd have to look more into how the occ affects things and changes over time to give a definite answer.

I'd recommend checking what the `occ` is before you reset them to 0, I'd expect them to be 0 already.